dm/tests/openapi/run.sh

```diff
@@ -358,6 +358,53 @@ function test_cluster() {
 	openapi_cluster_check "list_worker_success" 1
 }
 
+function test_noshard_task_dump_status() {
+	echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>START TEST OPENAPI: NO SHARD TASK DUMP STATUS"
+	prepare_database
+
+	task_name="test-no-shard"
+	target_table_name="" # empty means no route
+
+	# create source succesfully
+	openapi_source_check "create_source1_success"
+	openapi_source_check "list_source_success" 1
+
+	# get source status success
+	openapi_source_check "get_source_status_success" "mysql-01"
+
+	# create source succesfully
+	openapi_source_check "create_source2_success"
+	# get source list success
+	openapi_source_check "list_source_success" 2
+
+	# get source status success
+	openapi_source_check "get_source_status_success" "mysql-02"
+
+	# start task success: not vaild task create request
+	openapi_task_check "start_task_failed"
+
+	# start no shard task success
+	openapi_task_check "start_noshard_task_success" $task_name $target_table_name
+
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status $task_name" \
+		"\"unit\": \"Dump\"" 2
+
+	# check noshard task dump status success
+	openapi_task_check "check_noshard_task_dump_status_success" "$task_name" 0
+
+	init_noshard_data
+	check_sync_diff $WORK_DIR $cur/conf/diff_config_no_shard.toml
```

Comment:
I remember `dumpUnitProcessForever` will always be blocking, not dumping anything and waiting the task is canceled, so these data checking will not success. We can delete these lines.